handler/parse.go

```diff
@@ -11,56 +13,145 @@ type dataTree map[string]any
 type fileTree map[string]any
 
 // parseData parses incoming request body into data tree.
-func parseData(r *http.Request) dataTree {
+func parseData(r *http.Request) (dataTree, error) {
 	data := make(dataTree)
 	if r.PostForm != nil {
 		for k, v := range r.PostForm {
-			data.push(k, v)
+			err := data.push(k, v)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
 
 	if r.MultipartForm != nil {
 		for k, v := range r.MultipartForm.Value {
-			data.push(k, v)
+			err := data.push(k, v)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
 
-	return data
+	return data, nil
 }
 
 // pushes value into data tree.
-func (d dataTree) push(k string, v []string) {
+func (dt dataTree) push(k string, v []string) error {
 	keys := fetchIndexes(k)
 	if len(keys) <= MaxLevel {
-		d.mount(keys, v)
+		return dt.mount(keys, v)
 	}
+	return nil
+}
+
+func errWhenCreateNode(tree any, keys []string, value any) error {
+	return fmt.Errorf(
+		"invalid value in tree. key: %+v, val: %+v, tree: %+v",
+		keys[0],
+		value,
+		tree,
```

Comment:
The CI fails because you used a `tree` in the error message unescaped (CWE-117). You may have a look here at how to handle this: https://github.com/roadrunner-server/http/blob/master/middleware/log.go#L136.

But actually, I think that we should show the user the whole error message (which can contain sensitive data). We can log it and show a more generic error to the user.

Let's do the following. You can pass the logger from the handler (`h.log`). Then you'll need to unescape the tree using an approach I showed at the bottom.

So, the function `errWhenCreateNode` will be looking smt like this:

Reply:
yeah. I made it so now only the key is included in the error message.